docs/modules/ROOT/pages/udc.adoc

```diff
@@ -1,226 +1,144 @@
 
-To deploy a contract first you need to make sure it's been declared and then send a `deployContract` transaction to the Universal Deployer Contract.
-Here's an example implementation in Python:
+NOTE: The first iteration of the UDC is deployed at address `0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02bf` in Mainnet, Goerli testnet, Sepolia testnet, and starknet-devnet. See the {previous-udc-docs} for more information.
 
-[,python]
+== Interface
+
+[,javascript]
 ----
-from nile.common import UNIVERSAL_DEPLOYER_ADDRESS
-
-signer = MockSigner(PRIVATE_KEY)
-
-async def deploy_contract():
-    # Declare the contract and get the class hash before deployment
-    class_hash, _ = await signer.declare_class(account, "Ownable")
-
-    # Set up deployment parameters
-    salt = 123456  # random number
-    unique = FALSE
-    calldata = [account.contract_address]
-    params = [class_hash, salt, unique, len(calldata), *calldata]
-
-    # Send a "deployContract" transaction to the deployer contract
-    await account.send_transaction(
-        account,
-        UNIVERSAL_DEPLOYER_ADDRESS,
-        "deployContract",
-        params
-    )
+trait IUniversalDeployer {
+    fn deploy_contract(
+        class_hash: ClassHash,
+        salt: felt252,
+        from_zero: bool,
+        calldata: Span<felt252>
+    ) -> ContractAddress;
+}
 ----
 
-=== Deploying the UDC
-
-NOTE: The Universal Deployer Contract has already been deployed on most networks and development environments.
-The deployment of the UDC requires `deploy_from_zero=TRUE` to be passed to the `deploy` syscall. This results in a deterministic and predictable address across all instances of StarkNet, facilitating SDK integration and reproducibility of deployments.
-
-== Usage
+== Deploying a contract with the UDC
 
-The Universal Deployer Contract offers two types of addresses to deploy: deployer dependent and deployer independent.
-As the names suggest, the deployer-dependent type includes the deployer's address in the address calculation;
-whereas, the deployer-independent type does not.
-The `unique` boolean parameter ultimately determines the type of deployment.
+First, declare the target contract (if it's not already declared).
+Next, call the UDC's `deploy_contract` method.
+Here's an implementation example in Cairo:
 
-CAUTION: When deploying a contract that uses `get_caller_address` in the constructor calldata, remember that the UDC (and not the account) deploys that contract.
-Therefore, querying `get_caller_address` in a contract's constructor will return the UDC's address and *not* the account's address.
-
-=== Deployer dependent
+[,javascript]
+----
+#[starknet::interface]
+trait IUniversalDeployer<TContractState> {
+    fn deploy_contract(
+        ref self: TContractState,
+        class_hash: ClassHash,
+        salt: felt252,
+        from_zero: bool,
+        calldata: Span<felt252>
+    ) -> ContractAddress;
+}
```

Comment:
```suggestion
use openzeppelin::utils::interfaces::{IUniversalDeployerDispatcher, IUniversalDeployerDispatcherTrait};
```

Reply:
I think we should update the utils structure to match this URI, it is easier for users.

Reply:
It'd be structurally inconsistent to have `utils::interfaces` and `utils::cryptography::interface`, right? Would you agree that we should move `INonces` there as well?

Reply:
We can have a `utils::interfaces` reexporting the modules, we don't need to move the ones we already have, it is just for making the UX better.

Reply:
We can re-export INonces there as well.